src/test/java/com/amazon/ion/impl/IonRawTextWriterTest_1_1.kt

```diff
@@ -0,0 +1,1022 @@
+            stepOut()
+        }
+    }
+
+    @Test
+    fun `calling writeIVM when in a container should throw IonException`() {
+        assertWriterThrows {
+            stepInList(false)
+            writeIVM()
+        }
+    }
+
+    @Test
+    fun `calling writeIVM with a dangling annotation should throw IonException`() {
+        assertWriterThrows {
+            writeAnnotations(10)
+            writeIVM()
+        }
+    }
+
+    @Test
+    fun `calling finish should cause the buffered data to be written to the output stream`() {
+        val actual = writeAsString(autoClose = false) {
+            writeIVM()
+            finish()
+        }
+        // Just checking that data is written, not asserting the content.
+        assertTrue(actual.isNotBlank())
+    }
+
+    @Test
+    fun `after calling finish, it should still be possible to write more data`() {
+        val actual = writeAsString {
+            finish()
+            writeIVM()
+        }
+        // Just checking that data is written, not asserting the content.
+        assertTrue(actual.isNotBlank())
+    }
+
+    @Test
+    fun `calling close should cause the buffered data to be written to the output stream`() {
+        val actual = writeAsString(autoClose = false) {
+            writeIVM()
+            close()
+        }
+        // Just checking that data is written, not asserting the content.
+        assertTrue(actual.isNotBlank())
+    }
+
+    @Test
+    fun `calling close or finish multiple times should not throw any exceptions`() {
+        val actual = writeAsString {
+            writeIVM()
+            finish()
+            close()
+            finish()
+            close()
+            finish()
+        }
+        // Just checking that data is written, not asserting the content.
+        assertTrue(actual.isNotBlank())
+    }
+
+    @Test
+    fun `write the IVM`() {
+        assertWriterOutputEquals("\$ion_1_1") {
+            writeIVM()
+        }
+    }
+
+    @Test
+    fun `write nothing`() {
+        assertWriterOutputEquals("") {
+        }
+    }
+
+    @Test
+    fun `write a null`() {
+        assertWriterOutputEquals("null") {
+            writeNull()
+        }
+    }
+
+    @Test
+    fun `write a null with a specific type`() {
+        // Just checking one type. The full range of types are checked in IonEncoder_1_1Test
+        assertWriterOutputEquals("null.bool") {
+            writeNull(IonType.BOOL)
+        }
+    }
+
+    @ParameterizedTest
+    @CsvSource("true, true", "false, false")
+    fun `write a boolean`(value: Boolean, expected: String) {
+        assertWriterOutputEquals(expected) {
+            writeBool(value)
+        }
+    }
+
+    @Test
+    fun `write a delimited list`() {
+        assertWriterOutputEquals("[true,false]") {
+            stepInList(true)
+            writeBool(true)
+            writeBool(false)
+            stepOut()
+        }
+    }
+
+    @Test
+    fun `write a prefixed list`() {
```

Comment:
Do we really need variants for prefixed / delimited?

Reply:
I had included these test cases because the API allows you to specify prefixed / delimited (even if this implementation ignores that distinction). Is it really necessary? I don't know, but I don't see any harm in having them.

Reply:
That's fine. Knowing it was intentional is good enough for me.